lib/pages/logged_in/social_media_page/services/get_user_post_service.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class GetUserPostService {
+  final _firestore = FirebaseFirestore.instance;
+
+  Stream<QuerySnapshot> getPosts(List<String> friendUids) {
```

Comment:
Consider making a unifed class for generic CRUD operations with firebase/firestore.